adagrams/game.py

```diff
@@ -1,11 +1,134 @@
-    pass
+    #Goal: returns total score of a word
+    
+    #turn all characters in word into upper case like those in letter_bank
+    word = word.upper()
+    #build score chart as a dictionary
+    score_chart = {("A", "E", "I", "O", "U", "L", "N", "R", "S", "T"):1, ("D","G"):2, ("B", "C", "M", "P"):3, ("F", "H", "V", "W", "Y"):4, "K":5, ("J", "X"):8, ("Q", "Z"):10} 
+    
+    #initialize total score variable
+    score = 0
+    #loop thru each character in word
+    for char in word:
+        #loop thru each key in score_chart dict
+        for key in score_chart:
+            #if length of key is > 1. In other words, if there's > 1 letter for a point value...
+            if len(key) > 1:
+                #...then loop thru each letter in said key
+                for letter in key:
+                    #if character in word matches said letter...
+                    if char == letter:
+                        #...then add corresponding point value to word's score
+                        score += score_chart[key]
+            #if length of key is 1. In other words, if there's only 1 letter for a point value...            
+            else:
+                #...and if character in word matches said letter...
+                if char == key:
+                    #...then add corresponding point value to word's score
+                    score += score_chart[key]
+                    
+    #if the word is greater than 6 characters...
+    if len(word) > 6:
+        #...then add additional 8 pts to word's score
+        score += 8
+    
+    #return total score for word
+    return score   
+
+                
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #Goal: returns tuple containing winning word and highest score
+    # in the case of tie in scores, use these tie-breaking rules:
+    # prefer the word with the fewest letters...
+    # ...unless one word has 10 letters. If the top score is tied between multiple words and one is 10 letters long, choose the one with 10 letters over the one with fewer tiles
+    # If there are multiple words that are the same score and the same length, pick the first one in the supplied list
+    
+    #initialize winning_words_list list and word_score_dict dictionary
+    winning_words_list = []
+    word_score_dict = {}
+    #loop thru each word in list of words
+    for word in word_list:
+        #invoke function score_word to calculate score of each word looped thru
+        score = score_word(word)
+        #populate word_score_dict dictionary with word:score pairs
+        word_score_dict[word] = score
+    #get the word or words with the highest score from the dict and store in string_with_highest_score variable
+    string_with_highest_score = max(word_score_dict, key=word_score_dict.get)
+    #get the highest score from the dict
+    highest_score = word_score_dict[string_with_highest_score]
+    #print(highest_score) #verify
+    #loop thru dict
```

Comment:
Comments are good, and sometimes you can have too much of a good thing! In most of this function, your comments are almost exactly the same as the lines of code they document. We avoid that in industry because the lines of code may change, but devs aren't always careful about changing the comments to match. It's also more for humans to read, but doesn't give them extra information.

What comments might you write to describe *what* this code is doing without describing *how* it does it (since that is what's most likely to change later)?